also publish dev version on play store (aka nightlies)

3nids opened the pull request.

3nids commented on version.pri:
```diff
@@ -1,8 +1,9 @@
-VERSION_RC = 99
-VERSION_SUFFIX = ''
+# for RC (v1.2.3-rc4) or dev (v1.2.3-dev456)
+VERSION_NUMBER = 999
```
this has been increased to 3 digits, in case we have more than 99 commits on the dev (nightlies) build.
I guess it won't cause any trouble to the current build, since the global number will have 1 more digit (hence bigger).
Or is there a maximum value for this number (and I would be above now)
@mbernasocchi can you confirm?

mbernasocchi replied:
the main issue is that ve are getting one order of magnitude closer to the max version size.
https://stackoverflow.com/questions/24246143/maximum-length-of-android-versionname-versioncode-manifest

3nids commented on scripts/docker-build.sh:
```diff
@@ -44,17 +44,20 @@ if [[ "X${PKG_NAME}" != "Xqfield" ]]; then
+  sed -i "s/^VERSION_NUMBER\s*= .*/VERSION_NUMBER = ${VERSION_NUMBER:999}/" ${SOURCE_DIR}/version.pri
+
+  # coming fron env var, might be empty
+  sed -i "s/^VERSION_SUFFIX_STR\s*= .*/VERSION_SUFFIX_STR = ${VERSION_SUFFIX_STR}/" ${SOURCE_DIR}/version.pri
```
instead of being calculated here, it's defined ci/travis-build.sh:
https://github.com/opengisch/QField/pull/781/files#diff-e438ed61bbdfb465bfa5957775fa6912R20
and
https://github.com/opengisch/QField/pull/781/files#diff-e438ed61bbdfb465bfa5957775fa6912R27
so we have more flexibility (commit number added to version string)

3nids commented on scripts/deploy.sh:
```diff
@@ -3,30 +3,52 @@ set -e
 
 source scripts/travis_env.sh
 
+ARCHS=( armv7 x86 arm64_v8a x86_64 )
```
the main changes here are this array being added to avoid rewriting everywhere the same thing

3nids commented on scripts/deploy.sh:
```diff
@@ -3,30 +3,52 @@ set -e
+    done
+
+    echo -e "\e[93m * Deploying app to google play (release version)...\e[0m"
+    ./scripts/basic_upload_apks_service_account.py ch.opengis.qfield ${ASSETS}
```
here is the deploy to play store of the dev version

3nids commented:
> not really happy about adding one more 9 to the version number.

@mbernasocchi  OK, but I don't see many alternatives:

* switch to dates: `YYYYMMDDIA` (`I` an increment to allow up to 10 builds per day, `A` the architecture) but that bring us very close to the upper bound limit, no way to add one more digit
* totally rely on commit count to set the version number: this would be an easy way and will always increment. We just need to add a constant to be above current version. eg: `20000000 + commit_count`
* keep track of our own counter (with the same offset than previous solution)
* or the proposed 3 digits in the current PR, but only applied for the nightly builds

(current version is `10200995`)

qfield-fairy commented:
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-1f7426d1379ecfb93c770e5251e63abdcb291277-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-1f7426d1379ecfb93c770e5251e63abdcb291277-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-1f7426d1379ecfb93c770e5251e63abdcb291277-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-1f7426d1379ecfb93c770e5251e63abdcb291277-x86_64.apk)

mbernasocchi commented:
>     2\. totally rely on commit count to set the version number: this would be an easy way and will always increment. We just need to add a constant to be above current version. eg: `20000000 + commit_count`

why don't we just do the commit number? we can kill the current app id and start from 0 with a new id anyway. the package number is completely indipendent from the release one

3nids commented:
@mbernasocchi indeed, good point

m-kuhn commented on .travis.yml:
```diff
@@ -17,13 +17,9 @@ before_install:
-  - docker pull opengisch/qfield-sdk:${QFIELD_SDK_VERSION}
-  - echo "travis_fold:end:docker-pull"
-  - echo "travis_fold:start:tx-pull"
+  - export QFIELD_SDK_VERSION=$(cat sdk.conf | cut -d= -f2)
```
Why did you change this line?
The idea is to add versions for other platforms too, and we won't be able to rely on only one entry being in there.

3nids replied:
Nothing has changed to me here. Still looking in sdk.conf

m-kuhn replied:
But on the first line instead of the line for osgeo4a?

3nids replied:
Will revert

m-kuhn commented on scripts/docker-build.sh:
```diff
@@ -71,14 +74,14 @@ if [ -n "${KEYNAME}" ]; then
 	    --storepass "${STOREPASS}" \
 	    --keypass "${KEYPASS}" \
-        --input ${BUILD_DIR}/src/app/android-libqfield.so-deployment-settings.json \
+      --input ${BUILD_DIR}/src/app/android-libqfield.so-deployment-settings.json \
```
dedent?

qfield-fairy commented:
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-c57de4786427f296cd81d79842a3f3417b2d6c63-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-c57de4786427f296cd81d79842a3f3417b2d6c63-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-c57de4786427f296cd81d79842a3f3417b2d6c63-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-c57de4786427f296cd81d79842a3f3417b2d6c63-x86_64.apk)

3nids commented:
The main question is that I have removed the logic from version.pri and moved it to docker-build.sh
The logic was spread over Travis-build, docker-build and the version.pri, now it is a bit more centralized and more obvious I hope.
Does it work for you? 
Is it future proof for windows ? Or we don't care for now?

m-kuhn commented:
One thing I'm not sure is how it works for local builds. Setting the arch version is now in travis yml, so if we do a local build for a customer will there be an extra step manually?
Apart from this I'm not particularly attached to the old way of doing things, so if it's more centralized and still works, even better.

3nids commented:
It's all in docker-build, so pretty flexible and would work as is

qfield-fairy commented:
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-72756bd1d1c9a3fed028552279010da1ccae8190-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-72756bd1d1c9a3fed028552279010da1ccae8190-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-72756bd1d1c9a3fed028552279010da1ccae8190-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-72756bd1d1c9a3fed028552279010da1ccae8190-x86_64.apk)

3nids commented:
ok, now the method to determine the version code has been centralized here (and it's tested): https://github.com/opengisch/QField/pull/781/files#diff-d88da2f0ae98e992fab233374e8f3cb7

I also added a check to avoid creating too big version numbers
https://github.com/opengisch/QField/pull/781/files#diff-268efe2a5282875548af904d51a8577dR62
we can slowly increase it when needed

qfield-fairy commented:
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-939c8d47ae2d57364b03659301473199445194f6-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-939c8d47ae2d57364b03659301473199445194f6-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-939c8d47ae2d57364b03659301473199445194f6-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-939c8d47ae2d57364b03659301473199445194f6-x86_64.apk)

qfield-fairy commented:
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-86aeb881a10c974a94842470926e2d1f4ce6ee74-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-86aeb881a10c974a94842470926e2d1f4ce6ee74-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-86aeb881a10c974a94842470926e2d1f4ce6ee74-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-86aeb881a10c974a94842470926e2d1f4ce6ee74-x86_64.apk)

m-kuhn commented on scripts/docker-build.sh:
```diff
@@ -19,22 +26,17 @@ if [[ -z ${BUILD_FOLDER+x} ]]; then
-  PKG_NAME="qfield"
-fi
+
+[[ -z ${ARCH} ]] && ARCH=armv7
```
What was the  reason to change this?

m-kuhn replied:
```suggestion
# Set default values if missing
[[ -z ${ARCH} ]] && ARCH=armv7
```

m-kuhn commented on .travis.yml:
```diff
@@ -37,36 +35,39 @@ jobs:
         - docker-compose -f .docker/testing/docker-compose-travis.yml run qgis /usr/src/.docker/testing/build-test.sh
-      if: type = pull_request OR tag IS present OR branch = master
+
+    - stage: test
```
We are already now quite often short on parallel travis jobs, is it required to add one more?

3nids replied:
ok, I can drop it, although it lasts 40 sec

3nids replied:
done

m-kuhn commented on scripts/docker-build.sh:
```diff
@@ -44,17 +46,22 @@ if [[ "X${PKG_NAME}" != "Xqfield" ]]; then
+  echo "Generated version code: ${APP_VERSION_CODE}"
+fi
+
+# coming fron env var
```
```suggestion
# coming from env var
```

m-kuhn commented on scripts/version_number.sh:
```diff
@@ -0,0 +1,35 @@
+
+# version to version code
+# arg1: APP_VERSION (e.g. v1.2.3 or v1.2.3-rc4)
+# arg1: ARCH
```
```suggestion
# arg2: ARCH
```

m-kuhn commented on scripts/ci/travis_build.sh:
```diff
@@ -1,19 +1,56 @@
+
+  # adding the code name to the version string
+  CURRENT_CODENAME=$(cat ${DIR}/../../RELEASE_NAME)
+  APP_VERSION_STR="${TRAVIS_TAG} - ${CURRENT_CODENAME}"
```
The initial `v` needs to be stripped from TRAVIS_TAG

3nids replied:
fixed, use a dedicated and tested function

m-kuhn commented on RELEASE_NAME:
```diff
@@ -0,0 +1 @@
+Olavtoppen
```
This needs to be updated before every release, correct?

3nids replied:
yes, we could make it a bit more clever by looking in the release message, but it's quite some code and complexity. I don't think it's worth it.

m-kuhn replied:
Agreed, I thought the same, it was only to be sure :+1:

qfield-fairy commented:
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-421118ee53125e21009496abe8f8fffaaf454c80-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-421118ee53125e21009496abe8f8fffaaf454c80-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-421118ee53125e21009496abe8f8fffaaf454c80-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-421118ee53125e21009496abe8f8fffaaf454c80-x86_64.apk)

3nids commented on scripts/deploy.sh:
```diff
@@ -59,6 +59,6 @@ if [[ ${TRAVIS_SECURE_ENV_VARS} = true ]]; then
 
     echo -e "\e[93m * Deploying app to google play (release version)...\e[0m"
-    ./scripts/basic_upload_apks_service_account.py ch.opengis.qfield ${ASSETS}
+    ./scripts/basic_upload_apks_service_account.py ch.opengis.qfield-dev ${ASSETS}
```
@m-kuhn @mbernasocchi is this the channel we can push to?

m-kuhn replied:
sounds good to me :+1:

3nids replied:
do you want to wait for Marco's review or we merge and see what happens?

3nids replied:
(still need an approval before being able to merge)

qfield-fairy commented:
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-9e06b836ad5d0a78d95abbb40de9986bfd09b6c0-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-9e06b836ad5d0a78d95abbb40de9986bfd09b6c0-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-9e06b836ad5d0a78d95abbb40de9986bfd09b6c0-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-781-9e06b836ad5d0a78d95abbb40de9986bfd09b6c0-x86_64.apk)
